adxl345/adxl345.go

```diff
@@ -0,0 +1,252 @@
+	"fmt"
+	"periph.io/x/conn/v3/mmr"
+	"periph.io/x/conn/v3/physic"
+	"periph.io/x/conn/v3/spi"
+)
+
+type Sensitivity byte
+
+// The following constants are register used by the ADXL345
+// Check the table 19 of the datasheet for more details.
+// https://www.analog.com/media/en/technical-documentation/data-sheets/ADXL345.pdf
+const (
+	DeviceID = 0x00 // Device ID, expected to be 0xE5 when using ADXL345
+
+	ThreshTap        = 0x1D // Tap threshold
+	OfsX             = 0x1E // X-axis offset
+	OfsY             = 0x1F // Y-axis offset
+	OfsZ             = 0x20 // Z-axis offset
+	Dur              = 0x21 // Tap duration
+	Latent           = 0x22 // Tap latency
+	Window           = 0x23 // Tap window
+	ThreshAct        = 0x24 // Activity threshold
+	ThreshInact      = 0x25 // Inactivity threshold
+	TimeInact        = 0x26 // Inactivity time
+	ActInactCtl      = 0x27 // Axis control for activity/inactivity detection
+	ThreshFf         = 0x28 // Free-fall threshold
+	TapAxes          = 0x2A // Axis control for single tap/double tap
+	TapStatus        = 0x2B // Source of single tap/double tap
+	ActivityStatus   = 0x2A // Source of activity detection
+	InactivityStatus = 0x2B // Source of inactivity detection
+
+	// Control registers
+
+	BwRate     = 0x2C // Data rate and power mode control
+	PowerCtl   = 0x2D // Power saving features control
+	IntEnable  = 0x2E // Interrupt enable control
+	IntMap     = 0x2F // Interrupt mapping control
+	IntSource  = 0x30 // Source of interrupts
+	DataFormat = 0x31 // Data format control
+
+	// Data registers
+	DataX0 = 0x32 // X-Axis Data 0
+	DataX1 = 0x33 // X-Axis Data 1
+	DataY0 = 0x34 // Y-Axis Data 0
+	DataY1 = 0x35 // Y-Axis Data 1
+	DataZ0 = 0x36 // Z-Axis Data 0
+	DataZ1 = 0x37 // Z-Axis Data 1
+
+	// FIFO control
+	FifoCtl    = 0x38 // FIFO control
+	FifoStatus = 0x39 // FIFO status
+
+)
+
+// Sensitivity constants represents the sensitivity of the ADXL345.
+// The ADXL345 supports 4 sensitivity settings, ±2g, ±4g, ±8g, and ±16g, with the default being ±2g.
+// Sensitivity is an option that can be set at initialization in Opts.
+// You can set the sensitivity of the ADXL345 with the DataFormat register.
+const (
+	S2G  Sensitivity = 0x00 // Sensitivity at 2g
+	S4G  Sensitivity = 0x01 // Sensitivity at 4g
+	S8G  Sensitivity = 0x02 // Sensitivity at 8g
+	S16G Sensitivity = 0x03 // Sensitivity at 16g
+)
+
+var (
+	SpiFrequency = physic.MegaHertz * 2
+	SpiMode      = spi.Mode3 // Defines the base clock signal, along with the polarity and phase of the data signal.
+	SpiBits      = 8
+)
+
+var DefaultOpts = Opts{
+	ExpectedDeviceID: 0xE5,
+	Sensitivity:      S2G,
+}
+
+type Opts struct {
+	ExpectedDeviceID byte        // Expected device ID used to verify that the device is an ADXL345.
+	Sensitivity      Sensitivity // Sensitivity of the device (2G, 4G, 8G, 16G)
+}
+
+// Dev is a driver for the ADXL345 accelerometer
+// It uses the SPI interface to communicate with the device.
+type Dev struct {
+	c    mmr.Dev16
```

Comment:
Remove it if you don't plant to use it.

Reply:
Done ...  Will try to be a better citizen next time I will use mmr when providing I2C support. This version as been well tested on real hardwares via SPI only, i don't want to provide I2C without verifying it.